src/main/java/edu/umich/its/cpm/MigrationTaskService.java

```diff
@@ -1003,7 +1003,10 @@ private StringBuffer preMigrationChecks(StringBuffer itemStatus,
 				// return AsyncResult
 				return new AsyncResult<String>(setUploadJobEndtimeStatus(id, status));
 			}
-			fileName= replaceDotsInFileNameExceptFileExtention(fileName);
+			// replacing dots with _ don't need for Weblinks https://itsjira.umms.med.umich.edu/browse/TLCPM-653
+			if(!type.equals(Utils.CTOOLS_RESOURCE_TYPE_URL)) {
```

Comment:
fileName is now modified in 3 places: 1007, 1070, 1095.  Comments should explain why it needs to be done separately.  There may be good reasons but the reasons should be clear.

Is there a difference between CTools weblinks and box web_link_url?  Probably, but it should be clear in the code.  See lines 1007 to 1012.